app/Http/Controllers/ResumeController.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter,SlevomatCodingStandard.Functions.UnusedParameter
+
+namespace App\Http\Controllers;
+
+use App\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Storage;
+use App\Http\Requests\StoreResumeRequest;
+
+class ResumeController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('permission:read-users-resume|read-users-own', ['only' => ['showResume']]);
+        $this->middleware('permission:update-users-resume|update-users-own', ['only' => ['storeResume']]);
+        $this->middleware('permission:delete-users-resume|update-users-own', ['only' => ['deleteResume']]);
+        $this->middleware('permission:read-users-resume', ['only' => ['showResumeBook']]);
+    }
+
+    /**
+     * Show the user's resume.
+     *
+     * @param string $id
+     *
+     * @return \Illuminate\Http\JsonResponse|\Symfony\Component\HttpFoundation\BinaryFileResponse
+     */
+    public function showResume(string $id)
+    {
+        $user = User::findByIdentifier($id)->first();
+        if ($user) {
+            return response()->file(Storage::disk('local')->path('resumes/'.$user->uid.'.pdf'));
+        }
+
+        return response()->json(
+            [
+                'status' => 'error',
+                'message' => 'User does not exist or was previously deleted.',
+            ],
+            422
+        );
+    }
+
+    /**
+     * Show a resume book.
+     *
+     * @param string $datecode
+     *
+     * @return \Illuminate\Http\JsonResponse|\Symfony\Component\HttpFoundation\BinaryFileResponse
+     */
+    public function showResumeBook(string $datecode)
+    {
+        if (strlen(preg_replace('/[-0-9]/', '', $datecode)) > 0) {
+            return response()->json(
+                [
+                    'status' => 'error',
+                    'message' => 'invalid_datecode',
+                ],
+                400
+            );
+        }
+
+        return response()->file(Storage::disk('local')->path('resumes/resume-book-'.$datecode.'.pdf'));
+    }
+
+    /**
+     * Store the user's resume.
+     *
+     * @param string $id
+     * @param StoreResumeRequest $request
+     *
+     * @return \Illuminate\Http\JsonResponse|\Illuminate\Http\RedirectResponse
+     */
+    public function storeResume(string $id, StoreResumeRequest $request)
```

Comment:
Once the resume book function has been extracted to another controller, this should be renamed to remove `Resume` from the function name.